public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx

```diff
@@ -97,15 +103,62 @@ export const SaveDashboardAsForm = ({
     >
       {({ register, control, errors, getValues }) => (
         <>
-          <Field label="Dashboard name" invalid={!!errors.title} error={errors.title?.message}>
-            <Input
-              {...register('title', {
-                validate: validateDashboardName(getValues),
-              })}
-              aria-label="Save dashboard title field"
-              autoFocus
-            />
-          </Field>
+          <InputControl
+            render={({ field: { ref, ...field } }) => (
+              <Field
+                label={
+                  <HorizontalGroup justify="space-between">
+                    <Label htmlFor="title">Dashboard name</Label>
+                    {config.featureToggles.dashgpt && (
+                      <GenAIDashTitleButton onGenerate={(title) => field.onChange(title)} dashboard={dashboard} />
+                    )}
+                  </HorizontalGroup>
+                }
+                invalid={!!errors.title}
+                error={errors.title?.message}
+              >
+                <Input
+                  {...field}
+                  onChange={(e: ChangeEvent<HTMLInputElement>) => field.onChange(e.target.value)}
+                  aria-label="Save dashboard title field"
+                  autoFocus
+                />
+              </Field>
+            )}
+            control={control}
+            name="title"
+            rules={{
+              validate: validateDashboardName(getValues),
+            }}
+          />
+          <InputControl
+            render={({ field: { ref, ...field } }) => (
+              <Field
+                label={
+                  <HorizontalGroup justify="space-between">
+                    <Label htmlFor="description">Dashboard description</Label>
+                    {config.featureToggles.dashgpt && (
+                      <GenAIDashDescriptionButton
+                        onGenerate={(description) => field.onChange(description)}
+                        dashboard={dashboard}
+                      />
+                    )}
+                  </HorizontalGroup>
+                }
+                invalid={!!errors.description}
+                error={errors.description?.message}
+              >
+                <TextArea
+                  {...field}
+                  onChange={(e: ChangeEvent<HTMLTextAreaElement>) => field.onChange(e.target.value)}
+                  aria-label="Save dashboard description field"
+                  autoFocus
+                />
+              </Field>
+            )}
+            control={control}
+            name="description"
+          />
```

Comment:
yea, I agree. It does not make sense in this form

Reply:
and the amount additional markup/react code this needs is :exploding_head:

Reply:
My way of thinking was: Nowadays most dashboards doesn't contain description. When copying a dashboard for customizing it for your use case, could be a good moment to add it, if it doesn't have it already. This is why I didn't think on the description side.

@torkelo the amount of markup would be the same in case we skip the generation only for cloning since it will add only a conditional boolean. I have been thinking of different ways to reduce the markup code by avoiding this horizontal group to adapt it to the label, but any of them convinced me or covered all use cases.

Reply:
@ivanortegaalba I would start with having this only General settings title & description and gauge how many enable this feature / use it before adding it to other places. 

Usually, you use the "save as" action before you have made any changes (I guess it depends a bit, both scenarios are I guess common)

Reply:
> the amount of markup would be the same in case we skip the generation only for cloning since i

Yea try to find a way to reduce the complexity & amount of markup

Reply:
> @ivanortegaalba I would start with having this only General settings title & description and gauge how many enable this feature / use it before adding it to other places.

> Usually, you use the "save as" action before you have made any changes (I guess it depends a bit, both scenarios are I guess common)
Ok, yes. I was thinking about that yesterday. You might have a single dashboard before saving. On the other side is probably the first and only time you set the name of the dashboard. And about the description... I wonder how many dashboards really has description 😓 

I'm fine to close this PR and wait for adoption. @thanos-karachalios any thoughts on this?

Reply:
@torkelo what do you think about leaving the autogenerate only for the new dashboard and measure usage? It can help us to understand if it is useful or not. I'd not test something just for not getting the code dirty, I don't think it is that bad 😅